### PR TITLE
Fix programming errors in hr modules

### PR DESCRIPTION
**Pull request by ghost**
Fixed 2 programming errors in hr modules that make tests crash on other modules.

The first one is in hr_simplify. The file refers to the field 'state' on contract (hr.py line 159). But the field does not exist in this module or modules which it depends on.

The second error is in hr_job_category. It inherits the method write on contract and makes computations without validating the data passed to the method.

These modules obviously need unit tests.

### CODE REVIEW

File: hr_job_categories/hr.py [hunk 1]
```diff
@@ -107,23 +107,31 @@ def write(self, cr, uid, ids, vals, context=None):
...
-                    )
-                    self._remove_tags(
-                        cr, uid, contract.employee_id.id, prev_job_id,
-                        context=context
-                    )
-                    if vals.get('job_id'):
-                        self._tag_employees(
-                            cr, uid, contract.employee_id.id,
-                            contract.job_id.id, context=context
+                if(
```
Comment (pedrobaeza) [inline hr_job_categories/hr.py]: Why don't you simplify it doing `if vals.get('job_id') and data.get('id'):`?

Comment (ghost) [inline hr_job_categories/hr.py]: @pedrobaeza I agree. I pushed the change. Thanks!

**Comment by bwrsandman** [comment]
> These modules obviously need unit tests.

Okay, could you add two unittests?
- [x] `write()` of `hr.contract` in `hr_job_categories` when the condition you edited is true
- [x] for `_no_of_contracts()` in  `hr_simplify` which refer to state of `hr.contract`

**Comment by ghost** [comment]
@bwrsandman For now, I would prefer not to. I don't personally use nor inherit these modules but they interfere with my unittests in other modules in travis. If we begin to write unit tests in every untested modules from Michael Telahun Makonnen that generate bugs, I think it is going to take a lot of time.

**Comment by bwrsandman** [comment]
As a general rule, if you fix a bug, it is good practice to add a unittest to:
1. prove that the code works now
2. prevent regressions

No one is going to take it upon themselves to add unittests to every module in hr, but we can add them where needed.

I am only asking for two untests. One for each programming bug you found. Seems reasonable.

**Comment by ghost** [comment]
@bwrsandman ok, I'll put unittests there. Thanks

File: hr_simplify/tests/test_hr_simplify.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        and returns the number of employee
+        """
+        cr, uid, context = self.cr, self.uid, self.context
+
+        res = self.job_model._no_of_contracts(
+            cr, uid, [self.job_id],
+            name='no_of_employee', args=None, context=context
+        )
+
+        self.assertTrue(res[self.job_id]['no_of_employee'] == 1)
```
Comment (bwrsandman) [inline hr_simplify/tests/test_hr_simplify.py]: You can do `self.assertEqual(res[self.job_id]['no_of_employee'], 1)`, you'll get a more informative error message when it goes wrong.

**Comment by bwrsandman** [comment]
:+1: Very good.

**Comment by ghost** [comment]
@bwrsandman Thanks!

**Comment by max3903** [comment]
:+1:

**Pull request merged**